catalogue-service/src/main/java/eu/nimble/service/catalogue/persistence/util/CatalogueLinePersistenceUtil.java

```diff
@@ -41,8 +44,29 @@ public static CatalogueLineType getCatalogueLine(Long hjid) {
         return new JPARepositoryFactory().forCatalogueRepository().getSingleEntity(QUERY_GET_BY_HJID, new String[]{"hjid"}, new Object[]{hjid});
     }
 
-    public static List<CatalogueLineType> getCatalogueLines(List<Long> hjids) {
-        return new JPARepositoryFactory().forCatalogueRepository().getEntities(QUERY_GET_BY_HJIDS, new String[]{"hjids"}, new Object[]{hjids});
+    public static List<CatalogueLineType> getCatalogueLines(List<Long> hjids,CatalogueLineSortOptions sortOption,int limit, int offset) {
+
+        List<CatalogueLineType> catalogueLines = new ArrayList<>();
+        String getCatalogueLinesQuery = QUERY_GET_BY_HJIDS;
+        if(sortOption != null){
+            switch (sortOption){
+                case PRICE_HIGH_TO_LOW:
+                    getCatalogueLinesQuery += " ORDER BY cl.requiredItemLocationQuantity.price.priceAmount.value DESC NULLS LAST";
+                    break;
+                case PRICE_LOW_TO_HIGH:
+                    getCatalogueLinesQuery += " ORDER BY cl.requiredItemLocationQuantity.price.priceAmount.value ASC NULLS LAST";
+                    break;
+            }
+        }
+
+        catalogueLines = new JPARepositoryFactory().forCatalogueRepository().getEntities(getCatalogueLinesQuery, new String[]{"hjids"}, new Object[]{hjids});
+
+        if(limit != 0){
+            int startIndex = limit*offset;
```

Comment:
Let limit be 8 and offset be 1. If we have 10 catalogue lines as the result of the query,then this method should return 8 catalogue lines since the limit is 8. However, it seems that it will not return 8 catalogue lines because of the startIndex.

Reply:
@dogukan10  so normally offset starts from 0 know for pagination purposes in the front end.And yes if the limit is 8 and offset is 1 it will return only the last 2 products know ?.Offset is like the page no know ?